### DISCUSSION TITLE
Can't activate repo when automerge is set in branch mode

### DISCUSSION THREAD

**Discussion by EpicPlayerA10** [renovatebot/renovate#31875] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have automerge set to branch mode, so there isn't any PR created during updating libs, so how the repo can be activated? Shouldn't the repo activate automatically when the automerge happen?

![image](https://github.com/user-attachments/assets/d5f3b6b2-b3e2-4156-ab00-1e68e250032a)

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Merge one PR

**Reply by EpicPlayerA10** [reply]
Like, I can merge any PR? Not just these from renovate? I'm asking because i have configured renovate to use automerge branch mode which does not create any PR.

**Reply by rarkins** [reply]
A Renovate PR

**Reply by EpicPlayerA10** [reply]
Uh, yeah... There will be a problem with this because in my template in `renovate.json` i set `":automergeBranch"` preset which automerges all updates without creating any PR. So when I create a new repo from this template and when i install renovate on it, then I don't get any PR and the renovate is still in `onboarding` state.

**Reply by rarkins** [reply]
You keep typing "onboarding" but do you mean "onboarded"? Your screenshot shows "onboarded".

**Reply by EpicPlayerA10** [reply]
yeah, i was typing it wrong. I meant `onboarded`

**Reply by EpicPlayerA10** [reply]
So? Any ideas?

**Reply by rarkins** [reply]
This is expected behavior. Repos without a merged update PR are on a daily schedule.

**Reply by EpicPlayerA10** [reply]
But in my scenario, i have automerge set to branch mode, so there isn't any PR created during updating libs, so how the repo can be activated? Shouldn't the repo activate automatically when the automerge happen?

**Reply by rarkins** [reply]
Your welcome to search the code for "activated" to find the existing logic and contribute an improvement

**Reply by EpicPlayerA10** [reply]
So returning to the discussion (https://github.com/renovatebot/renovate/pull/31927#issuecomment-2408616869) :

My first idea to this was to directly modify the `repoIsActivated` on automerge, but i don't know if `repoIsActivated` option will persist between runs.

So I need to do the `hasCommittedTo` method per platform? Or maybe you have a better idea.

**Reply by rarkins** [reply]
I'm working on https://github.com/renovatebot/renovate/compare/main...feat/scm-stats

After it's merged then the "is activated" logic could be updated to return true if 2 or more Renovate commits are found in the last 90 days. This would mean it would go activated after the first branch automerge completes.

**Reply by EpicPlayerA10** [reply]
So? Is there any further work towards it?

**Reply by rarkins** [reply]
Unfortunately not. I found it added 100ms+ to runs, which really adds up with millions of jobs
